packages/engine/src/datastore/table/proxy.rs

```diff
@@ -83,34 +85,36 @@ pub struct BatchWriteProxy<K: Batch> {
 
 impl<K: Batch> BatchWriteProxy<K> {
     pub fn new(arc: &Arc<RwLock<K>>) -> Result<BatchWriteProxy<K>> {
-        if unsafe { arc.raw() }.try_lock_exclusive() {
+        // Safety: `try_lock_exclusive` locks the `RawRwLock` and returns `bool` instead of a guard.
+        if unsafe { RwLock::raw(arc).try_lock_exclusive() } {
             Ok(BatchWriteProxy { arc: arc.clone() })
         } else {
             Err(Error::from("Couldn't acquire exclusive lock on object"))
         }
     }
 
-    pub fn inner_mut(&mut self) -> &mut K {
+    pub fn batch(&self) -> &K {
         let ptr = self.arc.data_ptr();
         // Safety: This method uses unsafe code to get a mutable reference to the underlying data.
```

Comment:
This isn't a mutable reference anymore

Reply:
Git messed this up. I changed the name from `inner` to `batch` and `inner_mut` to `batch_mut`, but I also changed the order of `batch` and `batch_mut`, so git thinks, `inner_mut` got `batch` and `inner` got `batch_mut`.

Reply:
I have rewritten the `SAFETY`-sections in that file.